ru/lessons/advanced/otp-supervisors.md

```diff
@@ -0,0 +1,93 @@
+---
+layout: page
+title: OTP Супервизоры
+category: advanced
+order: 6
+lang: ru
+---
+
+Супервизоры это специальные процессы с единственной целью: мониторинг других процессов. Супервизоры позволяют нам создавать отказоустойчивые приложения при помощи автоматического перезапуска дочерних процессов в случае их отказа.
+
+## Содержание
+
+- [Настройка](#section-1)
+  - [Стратегии](#section-2)
+  - [Вложенность](#section-3)
+- [Наблюдатель задач](#section-4)
+  - [Настройка](#section-5)
+  - [Наблюдаемые задачи](#section-6)
+
+## Настройка
+
+Магия Супервизоров заключена в функции `Supervisor.start_link/2`.  Помимо запуска нашего Супервизора и его потомков, она позволяет нам определять стратегию, которую будет использовать наш Супервизор для управления дочерними процессами.
+
+Потомки определяются при помощи списка и функции `worker/3`, которую мы импортировали из `Supervisor.Spec`.  Функция `worker/3` принимает модуль, аргументы и набор опций.  Под капотом `worker/3` вызывает `start_link/3` с нашими аргументами во время инициализации.
+
+Давайте начнём, используя SimpleQueue из урока [OTP Concurrency](/lessons/advanced/otp-concurrency):
+
+```elixir
+import Supervisor.Spec
+
+children = [
+  worker(SimpleQueue, [], [name: SimpleQueue])
+]
+
+{:ok, pid} = Supervisor.start_link(children, strategy: :one_for_one)
+```
+
+Если наш процесс упадёт или завершится, Супервизор автоматически перезапустит его как ни в чём не бывало.
+
+### Стратегии
+
+На данный момент для Супервизора доступны четыре разные стратегии перезапуска:
+
++ `:one_for_one` - Перезапуск только упавшего дочернего процесса.
+
++ `:one_for_all` - Перезапуск всех дочерних процессов из события падения.
+
++ `:rest_for_one` - Перезапуск упавшего процесса и всех процессов, запущенных после него.
+
++ `:simple_one_for_one` - Наилучшая для динамически прикрепляемых процессов. Супервизор обязан содержать только одного потомка.
+
+### Вложенность
+
+Помимо воркеров, мы также можем наблюдать за другими Супервизорами, создавая дерево Супервизоров.  Единственное отличие для нас &mdash; замена `supervisor/3` на `worker/3`:
+
+```elixir
+import Supervisor.Spec
+
+children = [
+  supervisor(ExampleApp.ConnectionSupervisor, [[name: ExampleApp.ConnectionSupervisor]]),
+  worker(SimpleQueue, [[], [name: SimpleQueue]])
+]
+
+{:ok, pid} = Supervisor.start_link(children, strategy: :one_for_one)
+```
+
+## Наблюдатель задач
```

Comment:
I think it would be wise to add all Erlang/Elixir-specific + non-widely-used CS terms (tuple, for example, is crazy word).
From the Armstrong book translation I see that the first one is translated as "рабочий процесс", and the second one is "супервизор":

> Worker1 , Worker2 и т.д. это кортеж описывающий как запускать каждый из рабочих процессов. 

> Задача супервизора sellaprime - это конечно же держать всегда запущенными сервер простых чисел и сервер площади.